lib/kafka_ex/server.ex

```diff
@@ -281,9 +281,16 @@ defmodule KafkaEx.Server do
             :leader_not_available
           broker -> case produce_request.required_acks do
             0 ->  NetworkClient.send_async_request(broker, produce_request_data)
-            _ -> broker
-            |> NetworkClient.send_sync_request(produce_request_data, state.sync_timeout)
-            |> Produce.parse_response
+            _ ->
+              response = broker
+               |> NetworkClient.send_sync_request(produce_request_data, state.sync_timeout)
+               |> Produce.parse_response
+              case response do
+                [%KafkaEx.Protocol.Produce.Response{partitions: [%{error_code: error_code, offset: offset, partition: _}], topic: topic}] when (error_code == 0 or error_code == :no_error) and offset != nil ->
```

Comment:
`error_code` can only be `:no_error` see: https://github.com/kafkaex/kafka_ex/commit/37f6ddd9760d615b48e84214f0b86b9bc7d5e1b6